packages/protocol/contracts/rrp/authorizers/Api3TokenLock.sol

```diff
@@ -0,0 +1,304 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "./Api3RequesterRrpAuthorizer.sol";
+import "./interfaces/IApi3Token.sol";
+import "./interfaces/IApi3TokenLock.sol";
+import "../../admin/MetaAdminnable.sol";
+
+/// @title The contract used to lock API3 Tokens in order to gain access to Airnodes
+contract Api3TokenLock is MetaAdminnable, IApi3TokenLock {
+    enum AdminRank {
+        Unauthorized,
+        Admin,
+        SuperAdmin
+    }
+    uint256 private constant MAX_RANK = 2**256 - 1;
+    string private constant ERROR_ZERO_ADDRESS = "Zero address";
+    string private constant ERROR_ZERO_AMOUNT = "Zero amount";
+    string private constant ERROR_INSUFFICIENT_AMOUNT = "Insufficient amount";
+    string private constant ERROR_ALREADY_LOCKED = "Already locked";
+    string private constant ERROR_NOT_LOCKED = "No amount locked";
+    string private constant ERROR_LOCK_PERIOD_NOT_EXPIRED =
+        "Locking period not expired";
+    string private constant ERROR_REQUESTER_BLOCKED = "Requester Blocked";
+
+    /// @dev Address of api3RequesterRrpAuthorizer contract
+    address public immutable api3RequesterRrpAuthorizer;
+    /// @dev Address of Api3Token
+    address public immutable api3Token;
+    /// @dev The Minimum locking time (in seconds)
+    uint256 public minimumLockingTime;
+    /// @dev Lock amount for each user
+    uint256 public lockAmount;
+
+    /// @dev Represents the locked amounts, periods for the sponsors
+    /// and the total whitelist counts of a airnode-requester pair
+    struct AirnodeRequester {
+        mapping(address => uint256) sponsorLockAmount;
+        mapping(address => uint256) sponsorUnlockTime;
+        uint256 whitelistCount;
+    }
+
+    /// @dev Stores information about all token locks for airnode-requester pair
+    mapping(address => mapping(address => AirnodeRequester))
+        public airnodeToRequesterToTokenLocks;
```

Comment:
This mapping might not need to be public since we have getters for sponsorLockAmount and sponsorUnlockTime